StringOps.java

```diff
@@ -22,21 +22,133 @@ public class StringOps {
+    //System.out.println(capVowelsLowRest(s));        
+    System.out.println(camelCase (s));
+    //String string="Hello world";
+    //char chr='l';
+    //System.out.println(allIndexOf (string, chr));
+
     }
 
-    public static String capVowelsLowRest (String string) {
+    public static String capVowelsLowRest (String string) { //why doesnt it work when the first char is an uppercase vowel?
         // Write your code here:
-        return "";
+        String answer="";
+        for(int i=0;i<string.length(); i++)
+        {
+            if((string.charAt(i)=='a') || (string.charAt(i)=='e') || (string.charAt(i)=='i') || (string.charAt(i)=='o') || (string.charAt(i)=='u'))
+            {
+                if((string.charAt(i)>=(char)65) && (string.charAt(i)<=(char)90)) //not uppercase
+                {
+                    answer+=string.charAt(i);
+                }
+                else
+                {
+                    answer+=(char)(string.charAt(i)-32);
+                }
+            }
+            else
+            {
+                if((string.charAt(i)>=(char)65) && (string.charAt(i)<=(char)95))
+                {
+                    answer+=(char)(string.charAt(i)+32);
+                }
+                else
+                {
+                    answer+=string.charAt(i);
+                }
+            }    
+        }        
+
+        return answer;
+    }
+
+    public static boolean Cap(char c) //returns true if a char is uppercase
+    {
+        return c>='A' && c<='Z';
+    }
+
+    public static char upCase(char s) // returns an uppercase version of char 
+    {
+        if(s>='a' && s<='z')
+        {
+            return (char)(s-'a' + 'A');
+        }
+        else
+        {
+            return s;
+        }
     }
 
-    public static String camelCase (String string) {
+    public static char lowerCase(char s) // returns a lowercase version of char
+    {
+        if(s>='A' && s<='Z')
+        {
+            return (char)(s-'A'+'a');
+        }
+        else
+        {
+            return s;
+        }
+    }
+
+    public static String camelCase (String string) { 
         // Write your code here:
-        return "";
+        String answer="";
+        boolean cap=false;
+        boolean first=false;
+
+        for(int i=0; i<string.length(); i++)
+        {
+            if(!first && string.charAt(i)!=' ')
+            {
+                answer+=(lowerCase(string.charAt(i)));
+                first=true;
+                cap=false;
+                continue;
+            }
+
+            if(cap && string.charAt(i)!=' ')
+            {
+                answer+= upCase(string.charAt(i));
+                cap=false;
+            }
+            else
+            {
+                if(string.charAt(i)!= ' ' && !cap) 
+                {
+                    answer+= lowerCase(string.charAt(i));
+                }
+            }
+            if(string.charAt(i)==' ')
+            cap=true;
```

Comment:
always use `{}` to scope the body of your if statement - even when in include only one line of code.
```suggestion
            if(string.charAt(i)==' ') { 
               cap=true;
            }
```